spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1209,8 +1225,8 @@ private void commitIfNecessary() {
 				this.logger.debug("Commit list: " + commits);
 			}
 			if (!commits.isEmpty()) {
-				if (this.logger.isDebugEnabled()) {
-					this.logger.debug("Committing: " + commits);
+				if (LogSupport.isEnabled(this.commitLogLevel, ListenerConsumer.this.logger)) {
```

Comment:
Looks like duplicated code. More over I see that `commitLogLevel` is `final`. So, the decision about `isEnabled` we can make once in the ctor as well. Are there any reason do not? For example we change the logging level at runtime, so with my suggestion the current `ListenerConsumer` won't pick up a fresh config - only a fresh one after restarting container.

Just share that to be sure what we prefer: optimization over convenience or flexibility in the end.
Thanks

Reply:
Hmmm - they can't change the `commitLogLevel` dynamically but, let's say it's INFO and the category is currently WARN - they can change the category to INFO at runtime via a boot actuator.

Reply:
Right, that was my concern do not do that once. So, convenience is more important than optimization. OK, let it be.

May we consider to move this `isEnabled()` to the `LogSupport.Level` as well?

Reply:
Yes; I did that too - made more sense once I removed the statics.